spinn_front_end_common/interface/interface_functions/process_partition_constraints.py

```diff
@@ -28,12 +28,12 @@ class ProcessPartitionConstraints(object):
     :param ~pacman.model.graphs.application.ApplicationGraph application_graph:
     """
 
-    def __call__(self, machine_graph, application_graph=None):
+    def __call__(self, machine_graph):
         """
         :param ~.MachineGraph machine_graph:
         :param ~.ApplicationGraph application_graph:
         """
-        if application_graph is not None:
+        if machine_graph.application_level_used:
```

Comment:
Looking at this algorithm in particular, you probably don't even need to check; the code below checks the machine level and then only checks the application level 1) if the machine level doesn't have the interface and 2) if the machine level has an app level.  So this check, and the function "process_machine_partition" are both probably not needed...

Reply:
That was a choice between s slight speed increase vs algorithm complexity.

I was unsure and the their is a clear opinion for KISS, So will change

Reply:
Yep, that looks better :)